configure-mac_os_x.sh

```diff
@@ -126,13 +133,20 @@ cp -f mk/platform.mk.macosx $PLATFORM_MK
 cp -f mk/PcapPlusPlus.mk.common $PCAPPLUSPLUS_MK
 
 # set SDK home if MacOS verion >= 10.14
+MACOS_SDK_HOME="/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk"
 MACOS_MINOR_VERSION=`(sw_vers -productVersion) | awk -F '.' '{print $2}'`
 if [[ $MACOS_MINOR_VERSION -ge 14 ]]; then
    echo -e "\n# setting SDK home for MacOS version >= 10.14" >> $PCAPPLUSPLUS_MK
-   echo -e "MACOS_SDK_HOME := /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk\n" >> $PCAPPLUSPLUS_MK
+   echo -e "MACOS_SDK_HOME := $MACOS_SDK_HOME\n" >> $PCAPPLUSPLUS_MK
+fi
+
+PCAPPLUSPLUS_MK_MACOSX="PcapPlusPlus.mk.macosx"
+if [ -n "$BUILD_FOR_ARM64" ]; then
+   PCAPPLUSPLUS_MK_MACOSX="PcapPlusPlus.mk.macosx.arm64"
+   echo -e "MACOS_APPLE_SILICON_ARM64_BUILD_FLAGS := -arch arm64 -target arm64-apple-macos11 -isysroot $MACOS_SDK_HOME" >> $PLATFORM_MK
```

Comment:
👀  I am looking for a suggestion here : 

I am not really happy with the code duplication here.
( Lines copied from : https://github.com/seladb/PcapPlusPlus/blob/bd26f31b8229f0931542a6cba10436440d733cd9/mk/PcapPlusPlus.mk.macosx.arm64#L10 )

What would be the best way to clean it up?

Reply:
I'm currently adding support for Android and have the same issue:
https://github.com/seladb/PcapPlusPlus/blob/android-support/3rdParty/LightPcapNg/Makefile#L14-L19
https://github.com/seladb/PcapPlusPlus/blob/android-support/3rdParty/MemPlumber/Makefile#L9-L14
https://github.com/seladb/PcapPlusPlus/blob/android-support/3rdParty/hash-library/Makefile#L14-L19

I'm thinking that maybe we can add a new makefile where we can put these flags, and then include it in the 3rd-party Makefiles.
For example: we can add `3rdParty.mk.macosx.arm64` that includes these build flags and then pour its content to `3rdParty.mk` which we can include in the 3rd-party Makefiles.
Once you do this for Apple Silicon I can use the method for Android.
Please let me know what you think.